src/ConnectedMode.UnitTests/Migration/MigrationPromptTests.cs

```diff
@@ -0,0 +1,115 @@
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Linq;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Threading;
+using SonarLint.VisualStudio.ConnectedMode.Migration;
+using SonarLint.VisualStudio.Core;
+using SonarLint.VisualStudio.Core.Notifications;
+using SonarLint.VisualStudio.TestInfrastructure;
+
+namespace SonarLint.VisualStudio.ConnectedMode.UnitTests.Migration
+{
+    [TestClass]
+    public class MigrationPromptTests
+    {
+        [TestMethod]
+        public void MefCtor_CheckIsExported()
+        {
+            MefTestHelpers.CheckTypeCanBeImported<MigrationPrompt, IMigrationPrompt>(
+                MefTestHelpers.CreateExport<SVsServiceProvider>(),
+                MefTestHelpers.CreateExport<INotificationService>(),
+                MefTestHelpers.CreateExport<IThreadHandling>());
+        }
+
+        [TestMethod]
+        public void ShowAsync_VerifyNotificationIsCreatedWithExpectedParameters()
+        {
+            var notificationService = new Mock<INotificationService>();
+            INotification notification = null;
+            notificationService
+                .Setup(x => x.ShowNotification(It.IsAny<INotification>()))
+                .Callback((INotification n) => notification = n);
+
+            var serviceProvider = SetUpServiceProviderWithSolution("path_to_solution");
+
+            var testSubject = new MigrationPrompt(serviceProvider, notificationService.Object, new NoOpThreadHandler());
+            testSubject.ShowAsync().Forget();
+
+            notificationService.Verify(x => x.ShowNotification(notification), Times.Once);
+            notification.Id.Should().Be("migration_path_to_solution");
+            notification.Message.Should().Be(Resources.Migration_MigrationPrompt_Message);
+            notification.Actions.Count().Should().Be(2);
+
+            notification.Actions.First().CommandText.Should().Be(Resources.Migration_MigrationPrompt_MigrateButton);
+            notification.Actions.Last().CommandText.Should().Be(Resources.Migration_MigrationPrompt_LearnMoreButton);
+        }
+
+        [TestMethod]
+        public void ShowAsync_ShowNotificationIsCalledOnMainThread()
+        {
+            var notificationService = new Mock<INotificationService>();
+            var threadHandling = new Mock<IThreadHandling>();
+            Action runOnUiAction = null;
+            threadHandling
+                .Setup(x => x.RunOnUIThread(It.IsAny<Action>()))
+                .Callback((Action callbackAction) => runOnUiAction = callbackAction);
+
+            var serviceProvider = SetUpServiceProviderWithSolution();
+
+            var testSubject = new MigrationPrompt(serviceProvider, notificationService.Object, threadHandling.Object);
+            testSubject.ShowAsync().Forget();
+
+            threadHandling.Verify(x => x.RunOnUIThread(It.IsAny<Action>()), Times.Once);
+
```

Comment:
You could check that the `notificationService` hasn't been called at this point => checks that the thread switch happens first.